hmc7044: Fix support for Gekko solver

tachicialex opened the pull request:
Fix Gekko support for HMC7044. When applied in a HMC7044
test, Gekko solver would fail for too many times.
These changes made multiple plausible tests to pass.

Signed-off-by: Alexandru Tachici <alexandru.tachici@analog.com>

tfcollins commented on adijif/clocks/hmc7044.py:
```diff
@@ -278,15 +281,9 @@ def _get_clock_constraint(
-            od = self.model.Intermediate(eo * odd + (1 - eo) * even * 2)
+                even = self.model.Var(integer=True, lb=3, ub=2047)
+                odd = self.model.Intermediate(even * 2)
+                od = self.model.sos1([1, 2, 3, 4, 5, odd])
```
Is the spacing off here? These lines are unreachable

tfcollins commented on adijif/clocks/hmc7044.py:
```diff
@@ -27,7 +27,7 @@ class hmc7044(hmc7044_bf):
     # Defaults
     _d: Union[int, List[int]] = [1, 3, 5, *range(2, 4095, 2)]
-    _n2: Union[int, List[int]] = [*range(8, 65535 + 1)]
+    _n2: Union[int, List[int]] = [*range(8, 4095 + 1)]
```
Can we make this unique per solver? Since CPLEX should work over the extended ranges.

tfcollins commented on adijif/clocks/hmc7044.py:
```diff
@@ -16,8 +16,8 @@ class hmc7044(hmc7044_bf):
-    # n2_divider_max = 65535
-    n2_available = [*range(8, 65535 + 1)]
+    # n2_divider_max = 4095
+    n2_available = [*range(8, 4095 + 1)]
```
Can we make this unique per solver? Since CPLEX should work over the extended ranges.

tachicialex commented:
Changelog v1 -> v2:
 - fixed indent in _get_clock_constraint
 - added solver based n2 range selection:
     - gekko will use n2 = [8, 4095]
     - CPLEX will use n2 = [8, 65535]
